app/src/components/MessageList.tsx

```diff
@@ -77,9 +87,20 @@ class MessageList extends Component<Props, State> {
             } else {
                 this.fetchInitialMessages();
             }
+            this.handleScrollBackToSelectedMessage();
+        } else if ((messages[0] && !prevState.messages[0])
+            || (messages[0] && prevState.messages[0]
+                && (messages[0].timestamp !== prevState.messages[0].timestamp
+                    && messages[messages.length - 1].timestamp
+                    !== prevState.messages[prevState.messages.length - 1].timestamp))) {
```

Comment:
Is there a way to simplify this?

Reply:
If not can you write a comment on what it's checking for?

Reply:
sure, I'll write it out here and make sure it makes sense, and is what we want.

Reply:
1. Need to check for messages being given for the first time, where prevState would have an emprty array and current state is populated ( messages[0] && !prevState.messages[0] )
2. Need to make sure that when there is a new selected date and time (Currently the slider, but later graph would determine it) that it recenters the new selected message. We also need to make sure that it does not do this centering when it is simply new messages being added to the list of messages via the continuous scrolling.

Reply:
i think he means in code comment, which would require more concise explanation